src/main/java/ledger/user_interface/ui_controllers/MainPageController.java

```diff
@@ -28,6 +53,206 @@
     @FXML
     private Button addTransactionBtn;
 
+    // Transaction table UI objects
+    @FXML
+    private TableView transactionTableView;
+    @FXML
+    private TableColumn amountColumn;
+    @FXML
+    private TableColumn dateColumn;
+    @FXML
+    private TableColumn payeeColumn;
+    @FXML
+    private TableColumn typeColumn;
+    @FXML
+    private TableColumn categoryColumn;
+    @FXML
+    private TableColumn closedColumn;
+
+    // Transaction table edit event handlers
+    private EventHandler<CellEditEvent<TransactionModel, String>> amountEditHandler = new EventHandler<CellEditEvent<TransactionModel, String>>() {
+        @Override
+        public void handle(CellEditEvent<TransactionModel, String> t) {
+            try {
+                TransactionModel model = t.getTableView().getItems().get(t.getTablePosition().getRow());
+                String amountToSetString = t.getNewValue();
+                int dollarsToSet = Integer.parseInt(amountToSetString.substring(1, amountToSetString.length() - 3));
+                int centsToSet = Integer.parseInt(amountToSetString.substring(amountToSetString.length() - 2, amountToSetString.length()));
+                int amountToSet = (dollarsToSet * 100) + centsToSet;
+
+                Transaction transaction = model.getTransaction();
+                transaction.setAmount(amountToSet);
+
+                TaskWithArgs<Transaction> task = DbController.INSTANCE.editTransaction(transaction);
+                task.startTask();
+                task.waitForComplete();
+
+                updateTransactionTableView();
+            } catch (StorageException e) {
+                System.out.println("Error editing transaction amount: " + e.getMessage());
+            }
+        }
+    };
+
+    private EventHandler<CellEditEvent<TransactionModel, String>> dateEditHandler = new EventHandler<CellEditEvent<TransactionModel, String>>() {
+        @Override
+        public void handle(CellEditEvent<TransactionModel, String> t) {
+            try {
+                TransactionModel model = t.getTableView().getItems().get(t.getTablePosition().getRow());
+                String dateToSetString = t.getNewValue();
+
+                DateFormat formatter = new SimpleDateFormat("E MMM dd HH:mm:ss Z yyyy");
+                Date dateToSet = formatter.parse(dateToSetString);
+
+                Transaction transaction = model.getTransaction();
+                transaction.setDate(dateToSet);
+
+                TaskWithArgs<Transaction> task = DbController.INSTANCE.editTransaction(transaction);
+                task.startTask();
+                task.waitForComplete();
+
+                updateTransactionTableView();
+            } catch (StorageException e) {
+                System.out.println("Error editing transaction date: " + e.getMessage());
```

Comment:
Might wanna make an error popup for all of these (I won't tag each one so I don't spam everyone)